### DISCUSSION TITLE
Cannot proxy frozen objects

### DISCUSSION THREAD

**Discussion by dlindahl** [pmndrs/valtio#681] [Q&A] [answered]
This came about while attempting to put a GraphQL result into `valtio` with `@apollo/client`. When Apollo pulls from the cache in development mode, it freezes the result data to prevent accidental mutations that could leak back into the cache. When the result data is frozen, the browser throws a Proxy trap error:

```
TypeError: 'set' on proxy: trap returned truish for property 'test' which exists in the proxy target as a non-configurable and non-writable data property with a different value
```

...and points to [`vanilla.ts` line 231](https://github.com/pmndrs/valtio/blob/v1.10.2/src/vanilla.ts#L308) in the stack trace.

This is easy to reproduce with:

```
const obj = {}
Object.defineProperty(obj, 'test', {
  value: 123,
  writable: false,
  configurable: false,
})
const state = proxy(obj)
```

While I know I could probably deep clone the result data to work around this issue, it would be nice to retain Apollo's foot-gun protection logic. While I'm aware that there is some amount of object-freezing/property-defining work within `valtio` for performance reasons, its not clear to me if it'd be possible to natively support proxying frozen objects.

I did test using `ref` with the frozen object and while it does avoid throwing the trap error, I'm not sure its what I want since I _do_ want to track changes to the value when the Apollo cache propagates changes.

**Comment by dai-shi** [comment, marked as answer]
#673 should fix it.
Can you confirm with https://ci.codesandbox.io/status/pmndrs/valtio/pr/673 ?
(Find "Local Install Instructions".)

**Reply by dlindahl** [reply]
Can confirm that this PR fixes the problem! 🎉
